### PR TITLE
MGMT-15295: Fix bug in list manifests

### PR DESCRIPTION
**Pull request by paul-maidment**
While making a behavioural change to the way in which we list manifests https://github.com/openshift/assisted-service/commit/689a73c61b264d5e794d89e9a8f595a4ee920774 A bug was introduced. This bug causes a crash when manifests are listed.

This PR corrects the bug and introduces a test of the code in which the bug was found

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Review by javipolo — APPROVED** [review]

File: internal/manifests/manifests.go [hunk 1]
```diff
@@ -172,8 +172,6 @@ func (m *Manifests) ListClusterManifestsInternal(ctx context.Context, params ope
 		}
 		if isUserManifest {
 			manifests = append(manifests, &models.Manifest{FileName: filename, Folder: folder})
-		} else {
```
Comment (gamli75) [inline internal/manifests/manifests.go:175]: regarding line 154 - if the cluster doesn't exist please replace the "Object not found" message and use something like https://github.com/openshift/assisted-service/blob/master/internal/bminventory/inventory.go#L1535

**Review by filanov — APPROVED** [review]

**Comment by osherdp** [comment]
/retitle MGMT-15295: Fix bug in list manifests

**Comment by osherdp** [comment]
There's probably a condition affecting all sorts of jobs: https://redhat-internal.slack.com/archives/CBN38N3MW/p1689700194662809

**Pull request merged**